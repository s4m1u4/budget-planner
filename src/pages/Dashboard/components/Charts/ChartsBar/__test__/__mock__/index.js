export { TITLE, DATA } from "./mock";
