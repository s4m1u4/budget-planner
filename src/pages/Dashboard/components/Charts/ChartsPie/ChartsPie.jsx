import React from "react";
import { COLORS } from "../constants";
import { customLabel, customTooltip } from "../helpers";

import {
  Pie,
  Cell,
  Legend,
  Tooltip,
  PieChart,
  ResponsiveContainer,
} from "recharts";
import { ChartsPieTittle } from "./ChartsPie.styles";

export const ChartsPie = ({ title, data }) => {
  return (
    <div>
      <ChartsPieTittle>{title}</ChartsPieTittle>
      <ResponsiveContainer width="99%" height={300}>
        <PieChart>
          <Pie
            data={data}
            cx="50%"
            cy="50%"
            labelLine={false}
            outerRadius={80}
            fill="#8884d8"
            dataKey="value"
            label={customLabel}
          >
            {data.map((entry, index) => (
              <Cell
                key={`cell-${index}`}
                fill={COLORS[index % COLORS.length]}
              />
            ))}
          </Pie>
          <Legend iconType="circle" />
          <Tooltip content={customTooltip} />
        </PieChart>
      </ResponsiveContainer>
    </div>
  );
};
