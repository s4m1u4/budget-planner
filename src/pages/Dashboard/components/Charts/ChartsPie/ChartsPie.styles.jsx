import styled from "@emotion/styled";

export const ChartsPieTittle = styled.h4({
  fontSize: "20px",
  fontWeight: "500",
});
