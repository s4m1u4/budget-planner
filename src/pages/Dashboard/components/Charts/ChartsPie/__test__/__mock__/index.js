export { TITLE, DATA_INCOME } from "./mock";
