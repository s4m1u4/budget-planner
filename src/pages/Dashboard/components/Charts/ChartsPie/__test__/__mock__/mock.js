export const TITLE = "Income";
export const DATA_INCOME = [
  { name: "Work", value: 17369 },
  { name: "Travel", value: 106787 },
  { name: "Education", value: 10456 },
];
