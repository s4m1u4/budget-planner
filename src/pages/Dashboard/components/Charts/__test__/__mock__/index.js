export { CATEGORIES, RECORDS } from "./mock";
