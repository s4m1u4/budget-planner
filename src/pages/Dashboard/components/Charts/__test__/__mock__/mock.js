export const RECORDS = [
  {
    id: "9cc6c0d7-24e3-46aa-aa7c-73e86c1dd230",
    amount: "2354.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
  {
    id: "bc732adb-7b27-4d76-944c-4e7013eb97fa",
    amount: "836.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
  {
    id: "9410db09-8e1c-4c76-a6b3-e766d9b53ae5",
    amount: "2454.0000",
    type: "income",
    user_id: "fe719e6f-ac63-4630-9a70-2112037f9078",
    category_id: "084a3558-6b14-4e38-9d24-47e912021384",
  },
];

export const CATEGORIES = [
  {
    id: "084a3558-6b14-4e38-9d24-47e912021384",
    title: "Work",
    description: "Work",
    isDefault: false,
  },
  {
    id: "0ce314b9-a547-4f57-9d64-e466777ba8e9",
    title: "Travel",
    description: "awdawda",
    isDefault: false,
  },
  {
    id: "ea298d16-8bdb-4c5b-8198-6b2e5ce2f9a2",
    title: "Education",
    description: "Education it's so important for you!",
    isDefault: false,
  },
];
