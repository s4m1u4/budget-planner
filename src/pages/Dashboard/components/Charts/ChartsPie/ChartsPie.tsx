import React, { FC } from "react";
import currency from "currency.js";
import { COLORS } from "../constants";
import {
  Pie,
  Cell,
  Legend,
  Tooltip,
  PieChart,
  ResponsiveContainer,
} from "recharts";

import { ChartsPieTittle } from "./ChartsPie.styles";

interface IPayload {
  name: string;
  value: number;
}

interface ICustomTooltip {
  active: boolean;
  payload: IPayload[];
}

interface ICustomLabel {
  value: number;
}

interface IChartsPieData {
  name: string | undefined;
  value: number;
}

interface ChartsPieProps {
  title: string;
  data: IChartsPieData[];
}

export const ChartsPie: FC<ChartsPieProps> = ({ title, data }) => {
  const customTooltip = ({ active, payload }: ICustomTooltip | any) => {
    if (active) {
      const amount = payload[0].value;
      return (
        <div
          style={{
            color:
              sessionStorage.getItem("theme") === "light" ||
              sessionStorage.getItem("theme") === null
                ? "#000"
                : "#fff",
            padding: "5px",
            backgroundColor:
              sessionStorage.getItem("theme") === "light" ||
              sessionStorage.getItem("theme") === null
                ? "#fff"
                : "#303030",
            border: "1px solid #cccc",
            borderRadius: "6px",
          }}
        >
          <p>{`${payload[0].name}: ${currency(amount).format()}`}</p>
        </div>
      );
    }
    return null;
  };

  const customLabel = ({ value }: ICustomLabel) => currency(value).format();

  return (
    <div>
      <ChartsPieTittle>{title}</ChartsPieTittle>
      <ResponsiveContainer width="99%" height={300}>
        <PieChart>
          <Pie
            className="test"
            data={data}
            cx="50%"
            cy="50%"
            labelLine={false}
            outerRadius={80}
            fill="#8884d8"
            dataKey="value"
            label={customLabel}
          >
            {data.map((entry, index) => (
              <Cell
                key={`cell-${index}`}
                fill={COLORS[index % COLORS.length]}
              />
            ))}
          </Pie>
          <Legend iconType="circle" />
          <Tooltip content={customTooltip} />
        </PieChart>
      </ResponsiveContainer>
    </div>
  );
};
