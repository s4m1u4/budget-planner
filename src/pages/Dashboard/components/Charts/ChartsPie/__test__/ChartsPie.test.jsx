import { ChartsPie } from "../ChartsPie";
import { render } from "@testing-library/react";
import { DATA_INCOME, TITLE } from "./__mock__";

const renderComponent = () => {
  return render(<ChartsPie title={TITLE} data={DATA_INCOME} />);
};

describe("ChartsPie component", () => {
  it("snapshot matching", () => {
    const { baseElement } = renderComponent();
    expect(baseElement).toMatchSnapshot();
  });
});
