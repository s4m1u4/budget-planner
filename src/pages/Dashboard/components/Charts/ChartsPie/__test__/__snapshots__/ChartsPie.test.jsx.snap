// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartsPie component snapshot matching 1`] = `
<body>
  <div>
    <div>
      <h4
        class="css-sqlmve"
      >
        Income
      </h4>
      <div
        class="recharts-responsive-container"
        style="width: 99%; height: 300px;"
      />
    </div>
  </div>
</body>
`;
